source/common/network/address_impl.cc

```diff
@@ -90,7 +90,7 @@ Ipv6Instance::Ipv6Instance(const std::string& address) : Ipv6Instance(address, 0
 
 Ipv6Instance::Ipv6Instance(const std::string& address, uint32_t port) : InstanceBase(Type::Ip) {
   memset(&ip_.ipv6_.address_, 0, sizeof(ip_.ipv6_.address_));
-  ip_.ipv6_.address_.sin6_family = AF_INET;
+  ip_.ipv6_.address_.sin6_family = AF_INET6;
```

Comment:
Credit for spotting the bug goes to @mattklein123 how noticed it after my Ipv6Instance change was merged.